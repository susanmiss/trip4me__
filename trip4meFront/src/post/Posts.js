import React, { Component } from "react";
import { list } from "./apiPost";
import { Link } from "react-router-dom";


class Posts extends Component {
    constructor() {
        super();
        this.state = {
            posts: [],
            page: 1,
        };
    }

    loadPosts = page => {
        list(page).then(data => {
            if (data.error) {
                console.log(data.error);
            } else {
                this.setState({ posts: data.reverse() });
            }
        });
    };

    componentDidMount() {
        this.loadPosts(this.state.page);
    }

    renderPosts = posts => {
        return (
            <>
                <div className="itemList" id="portfolio">
                    {posts.map((post, i) => {
                        return (
                            <div className="Item portfolio-item img-fluid" style={{ display: 'block', marginRight: "auto", marginLeft: "auto", width: "800px", height: "100%" }} key={i}>

                                <Link
                                    className="portfolio-link" to={`/post/${post._id}`}
                                >
                                    <img
                                        style={{ width: "800px" }}
                                        className="img-fluid" src={`${process.env.REACT_APP_API_URL
                                            }/post/photo/${post._id}`}
                                    />
                                </Link>

                                <div className="portfolio-caption bg-light" >
                                    <h4 className="text-center">{post.title}</h4>
                                    < br />
                                    <p className="text-muted text-center pb-3">
                                    </p>
                                </div>

                            </div>


                        );
                    })
                    }

                </div >
            </>
        );
    };

    render() {
        const { posts, page } = this.state;
        return (
            <section>


                {/* <h2 className="mt-2">
                    {!posts.length ? "No posts yet!" : ""}
                </h2> */}
                < div >
                    {this.renderPosts(posts)}
                </div >

            </section >

        );
    }
}

export default Posts;