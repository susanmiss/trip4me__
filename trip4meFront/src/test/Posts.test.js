import React from 'react';
import ReactDOM from 'react-dom'
import {shallow} from 'enzyme'; 
import renderer from 'react-test-renderer';
import Posts from '../post/Posts'


const setup = (props={}, state=null) => {
    const wrapper = shallow(<Posts {...props} />)
    if( state ) wrapper.setState(state);
    return wrapper;
  }

  const findByTestAttr = (wrapper, val) => {
    return wrapper.find(`[data-test="${val}"]`);
  }

describe("<Posts />", () => {

  it('mounts to DOM', () => {
      const div = document.createElement('div');
      ReactDOM.render(<Posts />, div);
      ReactDOM.unmountComponentAtNode(div);
  });
  it('renders correctly', () => {
      const tree = renderer
          .create(<Posts />)
          .toJSON();
      expect(tree).toMatchSnapshot();
  });
  let wrapper;
  beforeEach(() => wrapper = shallow(<Posts />));
  
  it('Renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
  })
  
  it('should render a <div />', () => { 
      expect(wrapper.find('div').length).toEqual(3);
  });

  it('should render a <h2>', () => { 
    expect(wrapper.find('h2').length).toEqual(1);
  });


})