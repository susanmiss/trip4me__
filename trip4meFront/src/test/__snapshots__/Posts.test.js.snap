// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> Renders correctly 1`] = `ShallowWrapper {}`;

exports[`<Posts /> renders correctly 1`] = `
<section
  className="page-section"
  id="portfolio"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <h2
        className="mt-2"
      >
        No posts yet!
      </h2>
      <div
        className="container"
      >
        
        
      </div>
    </div>
  </div>
</section>
`;
